### PR TITLE
cloud: add statefulset config for EKS multi-region

### PR DESCRIPTION
**Pull request by taroface**
This StatefulSet config is meant for use with the EKS multi-region docs (WIP).

- Added an `initContainer` that determines the AZ of each pod.
- Added `namespace` field to fill in with a region-appropriate namespace.
- Modified `--join` and `--locality` in the start command for the user to customize with namespace/region names. I am unclear on whether the values here need to match the `namespace` defined for the cluster above.
- Added the memory/CPU requests & limits block that is in the single-region configs (but not the current multi-region config).
- Also added `dns-lb-eks.yaml` for creating an EKS network load balancer.
- Also added `configmap.yaml` for modifying the Corefile.

### CODE REVIEW

File: cloud/kubernetes/multiregion/cockroachdb-statefulset-secure-eks.yaml [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+          # Other nodes aren't able to look up the unqualified hostname.
+          - exec
+            /cockroach/cockroach
+            start
+            --logtostderr
+            --certs-dir /cockroach/cockroach-certs
+            --advertise-host $(hostname -f)
+            --http-addr 0.0.0.0
+            # TODO: Replace the placeholder values in --join and --locality with the namespace of the CockroachDB cluster in each region (e.g., us-east-1).
+            # --join cockroachdb-0.cockroachdb.<cluster-namespace-1>,cockroachdb-1.cockroachdb.<cluster-namespace-1>,cockroachdb-2.cockroachdb.<cluster-namespace-1>,cockroachdb-0.cockroachdb.<cluster-namespace-2>,cockroachdb-1.cockroachdb.<cluster-namespace-2>,cockroachdb-2.cockroachdb.<cluster-namespace-2>,cockroachdb-0.cockroachdb.<cluster-namespace-3>,cockroachdb-1.cockroachdb.<cluster-namespace-3>,cockroachdb-2.cockroachdb.<cluster-namespace-3>
```
Comment (taroface) [inline cloud/kubernetes/multiregion/cockroachdb-statefulset-secure-eks.yaml:253]: Taking into account [this comment](https://github.com/cockroachdb/docs/issues/3395#issuecomment-667215351) on the --join flag, am I safe to remove one node (e.g. `cockroachdb-2`) from each region here?

**Review by DuskEagle — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51775)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @taroface)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/51775#-MEQnFgwEXMO8uLMaXSM:-MEQnFgwEXMO8uLMaXSN:bpu5mtm) (no related file):*
The Kubernetes manifests look good to me, as long as there's other documentation to go along with it. It might be helpful to add a README about how to use these manifests. I realize there might be some text on the docs site, but if people find these manifests without going through the docs site, they might be confused on how to use them.

Relatedly, it might be nice to create a separate directory for the multiregion EKS manifests, to separate it from the existing manifests in this directory.

---
___
*[cloud/kubernetes/multiregion/cockroachdb-statefulset-secure-eks.yaml, line 253 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/51775#-MCrMPQpeDcRzA_1DvZm-r4-253:-MEQnOnL6pShdzSlP_MH:b-8a1eh0) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f273488cb7cde883b74263e5cbc5a531a26ba97/cloud/kubernetes/multiregion/cockroachdb-statefulset-secure-eks.yaml#L253)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Taking into account [this comment](https://github.com/cockroachdb/docs/issues/3395#issuecomment-667215351) on the --join flag, am I safe to remove one node (e.g. `cockroachdb-2`) from each region here?
</blockquote></details>

You're definitely safe to, though I also think it's fine to leave it.

**Comment by taroface** [comment]
@DuskEagle TFTR! I moved the files and added a README, also slightly updating the existing README in `multiregion`.

**Review by DuskEagle — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51775)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DuskEagle and @taroface)

___
*[cloud/kubernetes/multiregion/eks/README.md, line 73 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/51775#-MF7_aI79kpUo57VmOdu:-MF7_aI79kpUo57VmOdv:b-ziea2v) ([raw file](https://github.com/cockroachdb/cockroach/blob/83407f9849d5d9e912b8f149a92350ca9e8d7ca1/cloud/kubernetes/multiregion/eks/README.md#L73)):*
> ```Markdown
> - `region2` and `region3` with the namespaces in which the CockroachDB pods will run in the other 2 regions.
> 
> - `ip1`, `ip2`, and `ip3` with the IP addresses of the EKS instances in the region.
> ```

This should be the IP addresses of the Network Load Balancerss in the other regions, not the EKS VMs.

**Pull request merged**